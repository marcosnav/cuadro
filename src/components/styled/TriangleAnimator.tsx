import styled, { css, keyframes, StyledProps } from 'styled-components';

interface IProps {
  animate?: boolean;
  reversed?: boolean;
}

const translateDown = keyframes`
  0%, 30%, 100% {
    transform: translateY(0);
  }
  15%, 45% {
    transform: translateY(10px);
  }
`;

const translateUp = keyframes`
  0%, 30%, 100% {
    transform: translateY(0);
  }
  15%, 45% {
    transform: translateY(-10px);
  }
`;

const wrapAnimation = (props: StyledProps<IProps>) => {
  const animation = props.reversed ? translateUp : translateDown;
  return props.animate ? css`${animation} 2s ease infinite forwards;` : 'none;';
};

const TriangleAnimator = styled.div<IProps>`
  animation: ${(props) => wrapAnimation(props)}
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
`;

TriangleAnimator.displayName = 'TriangleAnimator';

export default TriangleAnimator;
