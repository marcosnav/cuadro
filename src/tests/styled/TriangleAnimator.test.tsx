import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import TriangleAnimator from './../../components/styled/TriangleAnimator';

describe('<TriangleAnimator />', () => {
  test('renders properly', () => {
    const component = mount(<TriangleAnimator />);
    expect(component).toHaveStyleRule('animation', 'none');
    expect(component).toMatchSnapshot();
  });

  test('animate: true', () => {
    const component = mount(<TriangleAnimator animate={true} />);
    expect(component).toHaveStyleRule('animation', expect.stringContaining('2s ease infinite forwards'));
    expect(component).toMatchSnapshot();
  });

  test('animate: true && reversed: true', () => {
    const component = mount(<TriangleAnimator reversed={true} animate={true} />);
    expect(component).toHaveStyleRule('animation', expect.stringContaining('2s ease infinite forwards'));
    expect(component).toMatchSnapshot();
  });
});
