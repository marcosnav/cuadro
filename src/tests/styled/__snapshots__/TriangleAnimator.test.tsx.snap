// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TriangleAnimator /> animate: true && reversed: true 1`] = `
.c0 {
  -webkit-animation: hRfQHy 2s ease infinite forwards;
  animation: hRfQHy 2s ease infinite forwards;
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
}

<TriangleAnimator
  animate={true}
  reversed={true}
>
  <div
    className="c0"
    reversed={true}
  />
</TriangleAnimator>
`;

exports[`<TriangleAnimator /> animate: true 1`] = `
.c0 {
  -webkit-animation: ouCfz 2s ease infinite forwards;
  animation: ouCfz 2s ease infinite forwards;
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
}

<TriangleAnimator
  animate={true}
>
  <div
    className="c0"
  />
</TriangleAnimator>
`;

exports[`<TriangleAnimator /> renders properly 1`] = `
.c0 {
  -webkit-animation: none;
  animation: none;
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
}

<TriangleAnimator>
  <div
    className="c0"
  />
</TriangleAnimator>
`;
